Settings.c

```diff
@@ -261,10 +261,10 @@ static void ScreenSettings_readFields(ScreenSettings* ss, Hashtable* columns, co
          memset(&ss->fields[j], 0, sizeof(ProcessField));
       }
       int id = toFieldIndex(columns, ids[i]);
-      if (id >= 0)
-         ss->fields[j] = id;
-      if (id > 0 && id < LAST_PROCESSFIELD)
-         ss->flags |= Process_fields[id].flags;
+      if (id < 0 || id >= LAST_PROCESSFIELD)
```

Comment:
This check will fail for dynamically added columns, e.g. with pcp-htop. Also why not make this validity check part of the post-conditions of `toFieldIndex` instead?